[candidate_parameters] Update TestPlan to include DoB and DoD tabs (#6234)
aces/Loris: modules/candidate_parameters/test/TestPlan.md

        * Error: date cannot be later than today
    * Consent to Study = No (changing/updating an already existing 'Yes' consent); Date (Withdrawal) of Consent = valid random date; Confirmation (Withdrawal) Date of Consent = same random date
        * No error
        * Make sure they update properly in the front-end and backend

### Date of Birth Tab

> 32. Check that date of birth can only be amended if user has candidate_dob_edit permission.

Reviewer: does the tab show to the user if they dont have the permission ?

Author: yes, however the field is disabled.

> 33. Change the date of birth and try to save it. Check that it is updated on front-end and back-end.

Reviewer: ```suggestion
33. Change the date of birth and try to save it. Check that it is updated on front-end and database.
```

Reviewer 2: maybe `back-end database` would be better?

34. Check that entering a date later than today gives a data entry error.
35. Check that the date stored matches the _DOB Format_ specified in the configuration module. 

### Date of Death Tab

> 36. Check that date of death can only be set if user has candidate_dod_edit permission.

Reviewer: does the tab show to the user if they dont have the permission ?

Author: yes, however the field is disabled.

> 37. Change the date of death and try to save it. Check that it is updated on front-end and back-end.

Reviewer: ```suggestion
37. Change the date of death and try to save it. Check that it is updated on front-end and database.
```

38. Check that entering a date of death before the date of birth gives a data entry error.

> 39. Check that entring a date later than today gives a date entry error.

Reviewer: ```suggestion
39. Check that entering a date later than today gives a data entry error.
```

40. Check that the date stored matches the _DOD Format_ specified in the configuration module.

> 41. Check that filling out any instrument with a _Date of Administration_ later than the DoD changes the age displayed to the candidate's age at death (in months).

Reviewer: this doesn't seem to belong here, it should be in the `instruments` test_plan which doesn't exist yet.

@PapillonMcGill when you start working on https://github.com/aces/Loris/issues/6293. keep that in mind
